 # Lab 4 - The Pendulum II

In [121]:
# Make sure to run this cell before running any other code cells
%reset -f
import numpy as np
import data_entry2

***Edit the following cell to provide the information requested:***

- Your name: Mary Helena Hosseini
- Your student number: 41004235
- Partner's name: Raminder Bal

## Part A - Intro and announcements

***TA grading feedback here:***
- Feedback: Please review the equation for calculating the uncertainty of T (period). Please do record your values under their calculations in markdown cell. Your understanding of t-score is great"

## Part B - t-score mini lesson

***Summarize any important notes from the class discussion***

- t-score Calculation Review: The t-score of the photo in the slide is roughly 1.414. Since the result is close to 1 and much lower than a 5, we cannot be sure that the results of the two curves are different given the relative uncertainty. Calculation of this t-score is in the cell below.

In [122]:
(45-35)/np.sqrt(5**2 + 5**2)

1.414213562373095

## Continuation of Notes: 
- If the t-score is close to 1, then we cannot be confident that the two data curves are different given the uncertainty
- If the t-score is close to 5, then we are confident that the two data curves are different given the uncertainty

### Results From Last Week:
- Other group with similar uncertainties as us got similar t-scores. We got an uncertainties of 0.00011 with a t-score of 0.39408 which falls into a similar range as the other groups. 
- A trend that we concluded from the graph was that groups with lower relative uncertainty, got very high t-scores therefore had higher confidence.

### Precision Axis:
- Answer is D, the quality of measurement is judged on how much the measurement was improved. Therefore all the figures could be an improvement from the first graph.
- t-score of 0.3 means you cannot be confident that there is any difference between the measurements therefore the answer is figure A.

## Part C - Improvement rounds (at least 2)

***Summarize any important notes from the class discussion and familiarization done in Part C of the student instructions***

- The formula for the uncertainty of period is $ u_T_Ave = \frac{\alpha}{N_trials \times M_cycles} $

p.s. for some reason the formula is not showing up :)

***Write some notes below detailing your initial measurement plan***

- Long Version: The initial measurement plan requires two people. One person will sit directly in front of the pendulum while the other person will pull the pendulum to line up it with the angle degrees. After which the person sitting in front of the pendulum will give the ok that the pendulum is correctly lined up and is read to start recording the time. Our plan is to measure 15 cycles of the pendulum swinging back and forth with 15 trials. This is an increase from our previous 10 trials and 10 cycles.

## Answer to Each Question From Instructions
- Which data taking variables (number of trials and/or number of swings) will you change?

We changed the number of swings from 10 to 15. We also changed the number of trials from 10 to 15.

- Why do you expect that these changes will improve your overall results? Make sure to reference uncertainty equations.

From the graph of results presented in class we knew that to achieve a high t-score, we need to attain low uncertainties. We also hypothesized that increasing the number of trials would be able to provide a more accurate measurement of time and period for both angles. 

- Anything else that you might change in how you take the data?

We ditched the lining up the pendulum with the a card method because we noticed that the pendulum would swing in different directions instead of a straight line. We also switched the spot of the two people so that one person can make sure the pendulum is lined up correctly while the other person is making sure that the pendulum is following a straight path. 

- How much time will it take to collect your sets of data for a full round? 

We think that it will take about 20 minutes. 

***Use the cell below to load in a blank spreadsheet to fill in with your initial round of data. (Note that you should have 2 columns of data, of equal length.)***

In [123]:
de1 = data_entry2.sheet("lab04_1")

Sheet name: lab04_1.csv


***Use the cell below for your analysis code. Insert additional cells below as needed. (Note that you can copy in and update your analysis code from lab03.)***

# Calculations for 10 degrees angle

In [124]:
t1Ave = np.mean(Time10Vec)
print("Average of t1 is", t1Ave)
ut1Ave = np.std(Time10Vec, ddof=1)/(np.sqrt(15))
print("Standard deviation of t1 is", ut1Ave)

T1Ave = np.mean(Period10Vec)
print("Average of period1 is", T1Ave)
uT1Ave = np.std(Period10Vec, ddof=1)/(np.sqrt(15))
print("Standard deviation of period1 is", uT1Ave)

t1Count = Time10Vec

relt1ave = ut1Ave/len(t1Count)
print("Relative uncertainty of t1 is", relt1ave)
relT1ave = uT1Ave/len(t1Count)
print("Relative uncertainty of period1 is", relT1ave)

Average of t1 is 26.959999999999997
Standard deviation of t1 is 0.046944749693757136
Average of period1 is 1.7973333333333334
Standard deviation of period1 is 0.0031296499795838116
Relative uncertainty of t1 is 0.003129649979583809
Relative uncertainty of period1 is 0.0002086433319722541


## Calculations for 20 degrees angle (Round 1)

In [125]:
t2Ave = np.mean(Time20Vec)
print("Average of t1 is", t2Ave)
ut2Ave = np.std(Time20Vec, ddof=1)/(np.sqrt(15))
print("Standard deviation of t1 is", ut2Ave)

T2Ave = np.mean(Period20Vec)
print("Average of period1 is", T2Ave)
uT2Ave = np.std(Period20Vec, ddof=1)/(np.sqrt(15))
print("Standard deviation of period2 is", uT2Ave)

t2Count = Time20Vec

relt2ave = ut1Ave/len(t2Count)
print("Relative uncertainty of t2 is", relt2ave)
relT2ave = uT1Ave/len(t2Count)
print("Relative uncertainty of period2 is", relT2ave)

Average of t1 is 27.254666666666665
Standard deviation of t1 is 0.027994330491771618
Average of period1 is 1.8169777777777782
Standard deviation of period2 is 0.0018662886994514445
Relative uncertainty of t2 is 0.003129649979583809
Relative uncertainty of period2 is 0.0002086433319722541


## t-score Calculation (Round 1)

In [126]:
tScore = np.abs(T2Ave-T1Ave)/np.sqrt(uT2Ave**2+uT1Ave**2)
print("The t-score value is", tScore)

The t-score value is 5.391103010420984


***Reflect on the results of your first round below***

### Meaning of t-score
We got a t-score close to 5 therefore we know that our values are distinct and different. 
### Final Conclusions of Round 1
From the results of round one and round two we can conclude that we are extremely confident that there is a difference because our t-score was 5.39, this is a very reliable measurement because we had really low relative uncertainties of 0.0002 and 0.0002, similar to Cristiana's group that we compared results with.

## Relative Uncertainty 
- Round One: 0.0002 and 0.0002
- From Other Group: 0.0005 and 0.0004
## t-score
- Round One: 5.391
- From the Other Group: 6.165

# Round Two

In [127]:
de1 = data_entry2.sheet("lab04_2")

Sheet name: lab04_2.csv


## Calculations for 10 degrees angle (Round 2)

In [128]:
t1Ave = np.mean(Time10Vec)
print("Average of t1 is", t1Ave)
ut1Ave = np.std(Time10Vec, ddof=1)/(np.sqrt(18))
print("Standard deviation of t1 is", ut1Ave)

T1Ave = np.mean(Period10Vec)
print("Average of period1 is", T1Ave)
uT1Ave = np.std(Period10Vec, ddof=1)/(np.sqrt(18))
print("Standard deviation of period1 is", uT1Ave)

t1Count = Time10Vec

relt1ave = ut1Ave/len(t1Count)
print("Relative uncertainty of t1 is", relt1ave)
relT1ave = uT1Ave/len(t1Count)
print("Relative uncertainty of period1 is", relT1ave)

Average of t1 is 32.50833333333334
Standard deviation of t1 is 0.02544056253745649
Average of period1 is 1.8051851851851852
Standard deviation of period1 is 0.0011959276228286735
Relative uncertainty of t1 is 0.0014133645854142495
Relative uncertainty of period1 is 6.644042349048185e-05


## Calculations for 20 degrees angle (Round 1)

In [129]:
t2Ave = np.mean(Time20Vec)
print("Average of t1 is", t2Ave)
ut2Ave = np.std(Time20Vec, ddof=1)/(np.sqrt(18))
print("Standard deviation of t1 is", ut2Ave)

T2Ave = np.mean(Period20Vec)
print("Average of period1 is", T2Ave)
uT2Ave = np.std(Period20Vec, ddof=1)/(np.sqrt(18))
print("Standard deviation of period2 is", uT2Ave)

t2Count = Time20Vec

relt2ave = ut1Ave/len(t2Count)
print("Relative uncertainty of t2 is", relt2ave)
relT2ave = uT1Ave/len(t2Count)
print("Relative uncertainty of period2 is", relT2ave)

Average of t1 is 32.72
Standard deviation of t1 is 0.02512712775109023
Average of period1 is 1.8178086419753083
Standard deviation of period2 is 0.001396292700525169
Relative uncertainty of t2 is 0.0014133645854142495
Relative uncertainty of period2 is 6.644042349048185e-05


## t-score Calculation (Round 2)

In [130]:
tScore = np.abs(T2Ave-T1Ave)/np.sqrt(uT2Ave**2+uT1Ave**2)
print("The t-score value is", tScore)

The t-score value is 6.8663810714246365


## YOUR SUMMARY OF RESULTS AND SUBMIT

- A brief summary of your key results and key comparisons in 1-2 sentences.

Our t-score increased from 5.391 to 6.866 when we increased our cycles from 15 to 18 and trials from 15 to 18. From the results of round one and round two we can conclude that we are extremely confident that there is a difference because our t-score was 5.391 and 6.866, this is a very reliable measurement because we had really low relative uncertainties of (0.0002 and 0.0002) and (0.0014 and 0.0014), similar to Cristiana's group that we compared results with (their values are specified down below).

- 1-2 sentences to describe and justify your major improvements and how they proved to be successful or unsuccessful. It may be helpful to use relative uncertainty to quantify your argument.

We decided to increase the cycles from 15 to 18 and the trials from 15 to 18 because we hypothesized that an increase in both trials and cycles will increase our accuracy in calculations. Based on the results from the t-scores we are not sure if this is successful. This is due to our relative uncertainties in round two increased compared to round one (0.0002 to 0.0014), and our t-scores (5.391 to 6.866).

- 1-2 sentences addressing the experimental goal and describing the conclusions you can make from the results of your comparisons. How do your results from this week compare to Lab 3?

From our comparison of t-scores in each round and of the other groups, we can conclude that the period of a pendulum depends on amplitude by comparing the period with initial release angles of 10° and 20°. Last we we concluded that the period a pendulum does not depend on amplitude because the t-score value that we got was very low, 0.39408, even thought our uncertainties were also low. This week we know that the conclusion from last week is incorrect due to us increasing our cycles, and trials and therefore having a higher t-score. 

## Relative Uncertainty 
- Round One: 0.0002 and 0.0002
- Round Two: 0.0014 and 0.0014
- From Other Group: 0.0005 and 0.0004
## t-score
- Round One: 5.391
- Round Two: 6.866
- From the Other Group: 6.165

# Submit

Steps for submission:

1. Click: Run => Run_All_Cells
2. Read through the notebook to ensure all the cells executed correctly and without error.
3. File => Save_and_Export_Notebook_As->HTML
4. Upload the HTML document to the lab submission assignment on Canvas.

In [131]:
# trick to display tables in html output
display_sheets()

Sheet: de1   File: lab04_2.csv


Time10            Period10 Time20            Period20
Units      s                   s      s                   s
0      32.37  1.7983333333333331  32.74   1.818888888888889
1      32.52  1.8066666666666669  32.92   1.828888888888889
2      32.47  1.8038888888888889  32.63  1.8127777777777778
3      32.52  1.8066666666666669  32.69   1.816111111111111
4      32.74  1.8038888888888889  32.48  1.8044444444444443
5      32.61  1.8116666666666665  32.69   1.816111111111111
6      32.43  1.8016666666666667  32.69   1.816111111111111
7      32.42   1.801111111111111  32.66  1.8144444444444443
8      32.61  1.8116666666666665  32.73  1.8183333333333331
9      32.32  1.7955555555555556  32.60  1.8111111111111111
10     32.59  1.8105555555555557  32.91  1.8283333333333331
11     32.41  1.8005555555555555  32.83   1.823888888888889
12     32.54  1.8077777777777777  32.72  1.8183333333333331
13     32.59  1.8105555555555557  32.76  1.8199999999999998
14     32.64  1.8133333333333335   32.8   1.822222222222222
15     32.41  1.8005555555555555  32.65  1.8138888888888889
16      32.5  1.8055555555555556  32.68  1.8155555555555556
17     32.46  1.8033333333333335  32.78  1.8211111111111111